Prog/se-vacant-lots-arlington.Rmd

```diff
@@ -0,0 +1,582 @@
+* Change `dwelling_type` values incorrectly recorded as "Multiple-family dwellings" to "Multiple-family"
+* If `height_ft` is missing, fill in with `height_stories`
+
+```{r height-problems}
+
+height_problems <- se %>% 
+  filter(str_detect(height_ft, "[:alpha:]"))
+
+se1 <- se %>% 
+  mutate(height_ft = ifelse(str_detect(height_ft, "[:alpha:]") == TRUE,
+                            NA,
+                            as.numeric(height_ft))) %>% 
+  mutate(special_exception = 1) %>% 
+  mutate(dwelling_type = ifelse(dwelling_type == "Multiple-family dwellings",
+                                "Multiple-family",
+                                dwelling_type)) %>% 
+  mutate(height_ft = ifelse(is.na(height_ft) & !is.na(height_stories),
+                            height_stories * 10,
+                            height_ft)) %>% 
+  filter(dwelling_type != "Hotel")
+  
+  
+```
+
+Check to make sure there are no codes in the special exceptions that are NOT in the by-right zoning code, due to being entered incorrectly.
+
+```{r check-se-code}
+
+se1 %>% 
+  filter(!zoning_code %in% zoningcode$zoning_code) %>% 
+  nrow()
+
+```
+Filter out multiple-family exceptions.
+
+```{r get-mf}
+
+semf <- se1 %>% 
+  filter(dwelling_type == "Multiple-family")
+
+```
+
+### Fill in missing variables in special exceptions
+
+If the special exceptions inherit the same stipulations as the by-right zoning code, the fields were left blank. Use `fill()` to fill in these gaps.
+
+Replace missing lot coverage maximums with 56%- this is to be used if there is no specific designation, per the Arlington County zoning code.
+
+```{r fil}
+
+mf_fill <- semf %>% 
+  bind_rows(filter(zoningcode,
+                   zoning_code %in% semf$zoning_code,
+                   dwelling_type == "Multiple-family")) %>% 
+  group_by(zoning_code, dwelling_type) %>% 
+  fill(lot_area_min, .direction = "down") %>% 
+  fill(lot_area_min, .direction = "up") %>% 
+  fill(lotarea_per_unit, .direction = "down") %>% 
+  fill(lotarea_per_unit, .direction = "up") %>%
+  fill(height_ft, .direction = "down") %>% 
+  fill(height_ft, .direction = "up") %>% 
+  fill(lot_coverage_max, .direction = "down") %>% 
+  fill(lot_coverage_max, .direction = "up") %>% 
+  mutate(lot_coverage_max = ifelse(is.na(lot_coverage_max),
+                                   .56,
+                                   lot_coverage_max))
+
+```
+
+There are four mixed-use zoning codes that only allow for multiple-family development via special exception. The stipulations for height and FAR vary based on the lot size. Additionally, there is no lot size per unit, we will instead rely on the height or FAR (whichever one is more restrictive) to determine the number of units. This will occur after the lots have been merged with the zoning codes.
+
+Join the multiple-family special exceptions to the by-right zoning code. This is the zoning code we will use for analysis.
+
+```{r zoning-recombine}
+
+zoningcode1 <- mf_fill %>% 
```

Comment:
checking on this join - mf_fill n=12 + zoningcode (n=63)  = zoningcode1 (n=71??)

Reply:
this join is only taking observations from the zoning code that also appear in the special exceptions and have a multiple-family dwelling type- so you end up with 6 from the by right zoning code, and 6 from the special exceptions. then use the codes from the by right to fill in the missing parameters from the special exceptions.

Reply:
nevermind, I see what you mean with the 71. so we take the 6 multifamily zoning codes from the special exceptions, add the other two zones (that are not in `mf_fill` because they have no counter part in the by right zoning code) and then add those 8 back to the by right zoning code. so 6+ 2 + 63 = 71.

Reply:
ok one more correction- ^ this still applies, but there are four matches between the two codes, and 4 that appear in only special exception. So 4 + 4 + 63 = 71.